internal/fetchclient/fetchclient.go

```diff
@@ -79,24 +80,29 @@ func (c *Client) Fetch(ctx context.Context, config *source.Config) (string, erro
 }
 
 func (c *Client) fetch(ctx context.Context, config *source.Config) (string, error) {
+	ignoreVersions := slicesext.ToStructMap(config.Source.IgnoreVersions)
```

Comment:
I know the intention is to support the "latest" non-ignored version, but picking 1.67.1 when 1.68.0 is available seems a bit backwards.

Reply:
> Should we lift this outside the fetch function to avoid implementing this in each fetcher?

I think for that to work, each fetcher would need to not return just the latest version but all available versions.

> I know the intention is to support the "latest" non-ignored version, but picking 1.67.1 when 1.68.0 is available seems a bit backwards.

They've published it but from release notes it sounds like they'd prefer users use 1.67.1 instead of 1.68.0.

Reply:
Yeah, ideally we'd have the fetchers return a list of all versions and we do post-processing on that list. Although there's a reason we did it this way, I suspect there's some optimization. 

Anyways, seems fine, just feels like this should be resolved upstream and not by us working around versioning issues and special-casing.

Reply:
> Yeah, ideally we'd have the fetchers return a list of all versions and we do post-processing on that list. Although there's a reason we did it this way, I suspect there's some optimization.

I think for some ecosystems you can easily grab the latest without fetching all versions.

> Anyways, seems fine, just feels like this should be resolved upstream and not by us working around versioning issues and special-casing.

I agree but this feels like it could potentially be useful in the future. I don't know if all package ecosystems support yanking releases (maybe just NPM/cargo?).

Reply:
Sgtm.

Go also has the concept of redact.